pkg/ingester/ingester_test.go

```diff
@@ -1984,7 +1989,10 @@ func Test_Ingester_LabelValues(t *testing.T) {
 		i.utilizationBasedLimiter = &fakeUtilizationBasedLimiter{limitingReason: "cpu"}
 
 		_, err := i.LabelValues(ctx, &client.LabelValuesRequest{})
-		require.EqualError(t, err, tooBusyError.Error())
+		stat, ok := status.FromError(err)
+		require.True(t, ok)
+		require.Equal(t, http.StatusServiceUnavailable, int(stat.Code()))
+		require.Equal(t, tooBusyErrorMsg, stat.Message())
```

Comment:
Nit: These checks repeat at different places. We could extract them into a `checkGrpcStatusError` method.